src/service.ts

```diff
@@ -221,20 +228,27 @@ export class DebugService implements IDebugger, IDisposable {
         );
       });
     }
-    this._model.breakpoints.restoreBreakpoints(bpMap);
+
+    // able resotre breakpoints after realod if had before breakpoints and enable lifecycle debugging
+    if (bpMap.size > 0 && autoStart) {
+      this.session.debuggedClients.add(this.session.client.name);
+    }
 
     const stoppedThreads = new Set(reply.body.stoppedThreads);
     this._model.stoppedThreads = stoppedThreads;
 
-    if (!this.isStarted() && (autoStart || stoppedThreads.size !== 0)) {
+    if (!this.isStarted() && autoStart) {
       await this.start();
     }
 
-    if (stoppedThreads.size !== 0) {
-      await this.getAllFrames();
-    } else {
-      this.clearModel();
-      this.clearSignals();
+    if (this.isLifeCycleOnKernelState()) {
```

Comment:
How about using an early return instead?

```typescript
if (!this.isLifeCycleOnKernelState()) {
   return;
}
```

So it's more readable, and the rest of the code won't even show in the diff anymore.